src/coreclr/src/nativeaot/Runtime/EHHelpers.cpp

```diff
@@ -390,6 +452,39 @@ Int32 __stdcall RhpHardwareExceptionHandler(UIntNative faultCode, UIntNative fau
             RhFailFast();
         }
 
+        translateToManagedException = true;
+    }
+    else if (faultCode == STATUS_ACCESS_VIOLATION)
+    {
+        // If this was an AV and code manager is null, this was an AV in unmanaged code.
+        // Could still be an AV in one of our assembly helpers that we know how to handle.
+        bool inWriteBarrierHelper = InWriteBarrierHelper(faultingIP);
+        bool inInterfaceDispatchHelper = InInterfaceDispatchHelper(faultingIP);
+
+        if (inWriteBarrierHelper || inInterfaceDispatchHelper)
+        {
+            if (faultAddress < NULL_AREA_SIZE)
+            {
+                faultCode = STATUS_REDHAWK_WRITE_BARRIER_NULL_REFERENCE;
```

Comment:
I'm considering to renaming this to something else, but don't have a good name.

Reply:
How about one of?
`STATUS_REDHAWK_HELPER_CODE_NULL_REFERENCE`
`STATUS_REDHAWK_NATIVE_HELPER_NULL_REFERENCE`
`STATUS_REDHAWK_UNMANAGED_HELPER_NULL_REFERENCE`
`STATUS_REDHAWK_MANUALLY_MANAGED_NULL_REFERENCE`

Reply:
Or just `STATUS_REDHAWK_NULL_REFERENCE` ?

Reply:
STATUS_REDHAWK_NULL_REFERENCE is already taken and apparently we need to be able to distinguish between these two cases in the frame iterator.

I like STATUS_REDHAWK_UNMANAGED_HELPER_NULL_REFERENCE. Thanks!